### PR TITLE
Fix accessibility of settings and system navigation

### LINKED ISSUE #12594: Keyboard navigations issues in new settings menu
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Before Home Assistant 2022.5.0, in Configuration it was possible to navigate using Tab and Shift+Tab and select what we want to configure, which is not the case now.
When going to Settings and press tab, focus go to Home Assistant Cloud which is OK but pressing Tab again leave the page so it is not possible to navigate between settings menu items.
Moreover, pressing up/down does not help as focus move alone so we have to be very quick to be able to select what we want before focus get placed to the beginning of the menu, which make impossible a first discover of the new menu for example.

### Describe the behavior you expected

Pressing Tab and Shift+Tab should allow us to navigate to something else than Home Assistant Cloud, we should be able to navigate in all the settings and if possible, up and down arrow in browse mode should not move the focus, in fact focus should not be changed alone

### Steps to reproduce the issue

1. Go to Settings
2. navigate in the settings menu using Tab and Shift+Tab

### What version of Home Assistant Core has the issue?

2022.5.0

### What was the last working version of Home Assistant Core?

2022.4.7 

### In which browser are you experiencing the issue with?

Firefox 100 and Google Chrome 101.0.4951.54

### Which operating system are you using to run this browser?

Debian Buster 10.12

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

The focus moving alone when using up and down arrow in browse mode (so the screen reader mode to use specifics navigation keys and read text using up and down arrow) might be the same problem I observe in #8414

### LINKED ISSUE #12782: Accessibility of UI components with hyperlinks
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

In various areas of the frontend, anchor elements are used within or around components which have another interactive role (e.g. tab, button, menu item).  This causes several accessibility issues:
- **Confusing/extra speech**: A screen reader might say "visited link tab selected integrations".
- **Extra tab stops**: One for the lin and one for the button, with unpredictable label announcements.
- **Breaks some functionality**: When wrapped around an `mwc-tab` or `mwc-listitem`, arrow key functionality to move to the linked items is broken and the wrong number of items is reported.

In many cases, this also violates the HTML and/or ARIA specifications (e.g. a `<button>` inside a `<a>>` is not valid).

Some notable places this occurs are:
1. Back button
2. More options button menu for integrations
3. Navigation tab bars on settings pages

### Describe the behavior you expected

- Per spec, interactive components should have only one role
- One tab stop per component
- No breaks in container's functionality

### Steps to reproduce the issue

1. Turn on a screen reader
2. Try to interact with affected mentioned components

### What version of Home Assistant Core has the issue?

2022.5.5

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox & Chrome (latest stable)

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

Probably in all cases, this can easily be fixed by removing the anchor element in favor of using `window.open` or `window.location` with a simple click event listener.

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

- Adds ARIA roles and labels using the `mwc-list` API
- Create `ha-list-item` by copying `ha-clickable-list-item` and removing the anchor and associated properties & styles (the "clickable" version can be removed after other instances are fixed too)
- Navigation is handled simply using `window.location.href`
- Handle the external app config case in the same function (seems much simpler this way)

See my comments on linked issues for more context on inaccessibility.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #12594 
- This PR is related to issue or discussion: #12782
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

File: src/components/ha-navigation-list.ts [hunk 1]
```diff
@@ -44,15 +49,20 @@ class HaNavigationList extends LitElement {
...
   }
 
-  private _entryClicked(ev) {
-    ev.currentTarget.blur();
+  private _handleListAction(ev: CustomEvent<ActionDetail>) {
+    const path = this.pages[ev.detail.index].path;
+    if (path.endsWith("#external-app-configuration")) {
+      this.hass.auth.external!.fireMessage({ type: "config_screen/show" });
+    } else {
+      window.location.href = path;
```
Comment (bramkragten) [inline src/components/ha-navigation-list.ts:64]: This causes a complete refresh of the page, it should use the `navigate` function.

**Comment by bramkragten** [comment]
The reason we have `clickable-list-item` and use regular anchors is that it can then be used by default browser functions, like open in a new tab, copy link, save link, etc.

File: src/components/ha-navigation-list.ts (hunk 1 above)
Comment (steverep) [reply src/components/ha-navigation-list.ts:64]: Okay I'll make that change.

**Comment by steverep** [comment]
@zsarnett  or @balloob can you examine this one while Bram is away?  I'd like to get this issue closed.

Please see the comments.  If everyone thinks it's critical to keep regular anchors in these cases, then I guess I'll try to mess with the components.

I do keep an eye on the activity in the Material repo, and it looks like they got smarter about having buttons with an `href` in Material 3, but that's probably a ways away for them to release and HA to integrate.

**Review by balloob — APPROVED** [review]
We added anchor tags because a) use the platform and b) the native browser features are very nice. However, the native browser features are a neat little extra. Accessibility is not.

I hope that the new md components will provide better accessibility and we can re-instate our anchor tags.

**Pull request merged**